cluster_utils/client/__init__.py

```diff
@@ -0,0 +1,348 @@
+from __future__ import annotations
```

Comment:
Maybe we can add a docstring here explaining that this package contains the client-side API (aka API for the job script). 

Also, as we seem to standardize on using the client-server terminology, we should probably add an explanation of this to the docs; maybe a section explaining the overall architecture would be nice.

Reply:
So far I'm actually not really consistent with the terminology.  I agree that we should standardize this.  I'm not sure if "server" is actually a good term, as it is not actually serving anything, it is only receiving.  But at least at the moment, I don't have a better idea.

Reply:
I added an "Architecture & Nomenclature" page to the docs